queries.sql

```diff
@@ -8,3 +8,65 @@ SELECT (name, escape_attempts) FROM animals WHERE weight_kg > 10.50;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.40 AND 17.30;
+
+BEGIN;
+UPDATE animals
+SET species = NULL;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon%';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals
+WHERE birthdate > '2022-01-01';
+SAVEPOINT SP1;
+UPDATE animals
+SET weight_kg = weight_kg * -1;
+SELECT * FROM animals;
+ROLLBACK TO SP1;
+SELECT * FROM animals;
+UPDATE animals
+SET weight_kg = weight_kg * -1
+WHERE weight_kg < 0;
+COMMIT;
+SELECT * FROM animals;
+
+BEGIN;
+SELECT COUNT(*) FROM animals;
+SELECT 
+   COUNT(escape_attempts) 
+FROM 
+   animals
+WHERE escape_attempts = '0';
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, COUNT(*)
+FROM animals
+WHERE escape_attempts > 0
+GROUP BY neutered;
+SELECT neutered,
+       MIN(weight_kg),
+       MAX(weight_kg)
+FROM animals
+GROUP BY neutered;
+SELECT neutered,
+       AVG(escape_attempts)
+FROM animals
+WHERE date_of_birth >= '1990-01-01' AND date_of_birth <= '2000-12-31'
+GROUP BY neutered;
```

Comment:
👍 You've got a strong SQL query there, but to fully meet the project requirements, you should make a slight adjustment:

- [ ] The SQL query you've provided retrieves data from the "animals" table, focusing on the "neutered" column and calculating the average number of "escape_attempts" for animals born between January 1, 1990, and December 31, 2000, grouped by the "neutered" column. This query provides valuable insights into the relationship between neutering status and escape behavior for animals within that date range. However, as per the project requirement, you should specifically answer the question: "What is the average number of escape attempts per animal type of those born between 1990 and 2000?" To meet this requirement, please update your query to provide the average number of escape attempts per distinct animal type (digimon, pokemon) for animals born between 1990 and 2000.
- [ ] If needed, kindly update the screenshot in the pull request (PR) description. Your cooperation is greatly appreciated.

Your commitment to aligning with the project requirements is greatly appreciated! Keep up the good work! 👏📊🐾